### ISSUE TITLE
What's the difference between .cns file and call.cns file generated by cnvkit.py batch?

### ISSUE THREAD

**Issue by sotaro-kanematsu** [etal/cnvkit#568]
I am trying to apply CNVkit to detection of sCNAs (somatic copy number alternations) with clinical samples.
Now, as a test I am trying to detect the sCNAs in HCC1395 cell line. I performed copy number calling pipeline (cnvkit.py batch), and finally obtatined two cnv file.
One is .cns file and the other is .call.cns file, which contains p_ttest row. I wonder what is the difference between them.
Those files are both output of cnvkit segment program?
I had the simulated data (contaminating normal cell), and want to modifiy the last call.
when I performed as a follow, I got call.cns file, however, the file format was different.
How should I perform copy number calling in order to obtain same format as call.cns file obtained from defalut workflow?
    
"cnvkit.py call *.cns --purity 0.X -m threshold *.call.cns"

**Comment by tetedange13** [comment]
Hi @sotaro-kanematsu,
<br>

I am **NOT** an author of CNVkit, but I had the same question as yours.
Here are some elements I **deduced myself** based on the code of [`batch.py`](https://github.com/etal/cnvkit/blob/d0bea38da69bd666cd959347adcaf34b9d4bb223/cnvlib/batch.py) (latest releave 0.9.8)

- [Segmentation](https://cnvkit.readthedocs.io/en/stable/pipeline.html#segment) step starts (line 186) and the resulting Python object is given to an internal `segmetrics`, to add the "ci" statistic (confidence interval, see [here](https://cnvkit.readthedocs.io/en/stable/reports.html?highlight=ci#segmetrics)) 
=> *There you have your 1st ".cns" file (not the ".call.cns" one) which is written*
=> It could be described as "raw", because no "segment filtering" nor "transformation" is applied on segmentation results (appart from adding the "CI" statistic)

- Then a "[call](https://cnvkit.readthedocs.io/en/stable/pipeline.html#call)" step starts (line 201), based on the ".cns" generated at previous step
=> I let you read the doc, but aim here is to [filter](https://cnvkit.readthedocs.io/en/stable/pipeline.html#filtering-segments) "raw" segments to remove likely false positives and then calling the copy_number as an integer
=> So segments are filtered on their "CI" statistic (calculated at previous step), then a [transformation](https://cnvkit.readthedocs.io/en/stable/pipeline.html?highlight=center#transformations) is applied to median-center log2 values and eventually copy_number call is done, based on threshold values (see [table here](https://cnvkit.readthedocs.io/en/stable/pipeline.html?highlight=center#calling-methods))
=> *In the meantime the "p_ttest" statistic you noticed is calculated and the 2nd ".call.cns" file is written*

To sum up:
- ".cns" = raw segmentation results
- ".call.cns" = filtered + called segmentation results

So an answer to your question would be:
1. Run your call: `call *.cns --purity 0.X --filter ci -m threshold > *.tmp.cns`
2. Then simply add the wanted statistic: `segmetrics -s *.tmp.cns --t-test > *.call.cns` 

<br>
Hope I was clear enough with my explanations.<br>
See you<br>
Felix !

**Comment by sotaro-kanematsu** [comment]
I am waiting for your reply (^^).
Yes, I got all.
When I ran a call considering tumor purity(used call.cns file as input), I saw CN showed negative integer value (-1,-2..).
OK, I try again correctly. 
Thank you
tetedange13

**Comment by tskir** [comment]
@tetedange13 Thank you for the great explanation! Indeed, `cnvkit.py segment` produces `*.cns` files, and `cnvkit.py call` produces `*.call.cns` files which have an added `cn` column.

However, @sotaro-kanematsu, I think your original command for calling is also correct, and it shouldn't have resulted in negative copy number values. Are you still observing this problem? If so, could you please share an example of a file where it occurs? (Just one line would be enough to demonstrate the issue, there's no need to share the full file if it can't be done.)

**Comment by sotaro-kanematsu** [comment]
Hi @tskir
Now, I have not resulted in negative copy number values with @tetedange13 suggestion.
I want to ask about the calling method. 
If I want to get allele frequencies with vcf files as input, what kind of vcf files are needed?
The manual advises me to use vcf file containing SNV calls for the tumor sample (variant calls file of HCC1395 is the thing for me), however it is unclear that  I give somatic mutations list (for example, output file obtained via somatic mutations caller such as mutect2), or just mutations list in tumor sample (for example, output file obtained via basic mutations caller such as bcftools).

**Comment by etal** [comment]
Thoroughly answered by contributors. Closing as answered — please reopen, or open a new issue, if a concrete problem remains.

For general usage questions, we now use GitHub Discussions (Q&A): https://github.com/etal/cnvkit/discussions
